File: src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionCsvParseTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;
public class TransformerFunctionCsvParseTest extends BaseTest {
    @Test
    void inline() {
        assertTransformation("a\n\",\"", "$$csvparse:$",
                             fromJson("""
                                [{ "a": "," }]"""));
        assertTransformation("a\n\"\"\"\"", "$$csvparse:$",
                             fromJson("""
                                              [{ "a": "\\"" }]"""));
        assertTransformation("1,2\n3,4", "$$csvparse(true):$",
                             fromJson("""
                                              [["1","2"],["3","4"]]"""));
    }

    @Test
    void object() {
        assertTransformation("a\n\",\"",
                             fromJson("""
                                {
                                  "$$csvparse": "$"
                                }"""),
                             fromJson("""
                                [{ "a": "," }]"""));
        assertTransformation("a\n\"\"\"\"",
                             fromJson("""
                                {
                                  "$$csvparse": "$"
                                }"""),
                             fromJson("""
                                              [{ "a": "\\"" }]"""));

    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionCsvParse.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionCsvParse", () => {
  test("inline", async () => {
    await assertTransformation('a\n","', "$$csvparse:$", [{ a: "," }]);
    await assertTransformation('a\n""""', "$$csvparse:$", [{ a: '"' }]);
    await assertTransformation("1,2\n3,4", "$$csvparse(true):$", [
      ["1", "2"],
      ["3", "4"],
    ]);
  });

  test("object", async () => {
    await assertTransformation(
      'a\n","',
      {
        $$csvparse: "$",
      },
      [{ a: "," }],
    );
    await assertTransformation(
      'a\n""""',
      {
        $$csvparse: "$",
      },
      [{ a: '"' }],
    );
  });
});
